app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\LoginRequest;
+use Illuminate\Support\Facades\Auth;
+
+class AuthController extends Controller
+{
+    public function login(LoginRequest $request)
+    {
+        $validator = $request->validated();
+
+        $field = filter_var($request->input('usernameOrEmail'), FILTER_VALIDATE_EMAIL) ? 'email' : 'username';
+        $request->merge([$field => $request->input('usernameOrEmail')]);
+        $credentials = $request->only($field, 'password');
```

Comment:
Chỉ có 2 field thì có thể lấy ra luôn, ko cần phải merge rồi lại gọi only
```suggestion
        $usernameOrEmail = $request->input('usernameOrEmail');
        $field = filter_var($usernameOrEmail, FILTER_VALIDATE_EMAIL) ? 'email' : 'username';
        $credentials = [$field => $usernameOrEmail, 'password' => $request->input('password')];
```

Reply:
vâng ạ